java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -582,14 +696,15 @@ private boolean isAcceptableByAutoboxing(JavaType expressionType, JavaType forma
 
   /**
    * JLS7 15.12.2.5. Choosing the Most Specific Method
+   * @param typeSubstitution 
    */
-  private JavaSymbol selectMostSpecific(JavaSymbol m1, JavaSymbol m2) {
+  private JavaSymbol selectMostSpecific(JavaSymbol m1, TypeSubstitution typeSubstitution, JavaSymbol m2) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "selectMostSpecific" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)